node/node.go

```diff
@@ -953,6 +954,77 @@ func (n *Node) GetPeerInfo(pid string) ([]core.QueryP2PPeerInfo, error) {
 	return peerInfoSlice, nil
 }
 
+// GetEpochStartDataForEpoch returns epoch start data of a given epoch
+func (n *Node) GetEpochStartDataForEpoch(epoch uint32) (*common.EpochStartDataAPI, error) {
+	if epoch == 0 {
+		// for the first epoch, epoch start identifier isn't committed. Therefore, return the genesis info
+		genesisHeader := n.dataComponents.Blockchain().GetGenesisHeader()
+		return &common.EpochStartDataAPI{
+			Nonce:     genesisHeader.GetNonce(),
+			Round:     genesisHeader.GetRound(),
+			ShardID:   genesisHeader.GetShardID(),
+			Timestamp: int64(time.Duration(genesisHeader.GetTimeStamp())),
+		}, nil
+	}
+
+	if n.bootstrapComponents.ShardCoordinator().SelfId() == core.MetachainShardId {
+		return n.getMetaFirstNonceOfEpoch(epoch)
+	}
+
+	return n.getShardFirstNonceOfEpoch(epoch)
+}
+
+func (n *Node) getShardFirstNonceOfEpoch(epoch uint32) (*common.EpochStartDataAPI, error) {
+	storer := n.dataComponents.StorageService().GetStorer(dataRetriever.BlockHeaderUnit)
+	if check.IfNil(storer) {
```

Comment:
So this condition is possible.

Reply:
Perhaps in a future PR we can adjust `GetStorer()` to return an error, as well (if storer not found).

Reply:
Already done in rc/1v.4.0:
- https://github.com/ElrondNetwork/elrond-go/blob/rc/v1.4.0/dataRetriever/chainStorer.go#L34

Reply:
Then let's add a TODO here to remove this check.IfNil

Reply:
added todo